examples/multi-tenant/src/collections/Users/access/updateAndDelete.ts

```diff
@@ -15,9 +17,22 @@ export const updateAndDeleteAccess: Access = (args) => {
 
   const adminTenantAccessIDs = getTenantAdminTenantAccessIDs(req.user)
 
-  return {
-    'tenants.tenant': {
-      in: adminTenantAccessIDs,
-    },
+  if (!data) {
+    // Triggered by visit to the User collection on Admin console
+    // If we return true, then this enables the "Create New" button
+    // so, we do that if we're an admin of at least 1 tenant
+    return Boolean(adminTenantAccessIDs.length)
   }
+
+  // If you're only a tenant-admin, then you can't delete super-admins
+  if (data.roles?.includes('super-admin')) {
+    return false
+  }
+
+  // Can delete a user, only if you're an admin in *all* of their tenants
+  return data.tenants?.every((tenant) =>
```

Comment:
What if I'm a tenant admin and I need to remove admin access for another user who is also an admin of another?

Reply:
This is a use-case not covered, yet, and talking about it has made me remember that this change would allow a tentant-admin to delete a user provided that they aren't also a super-user and also not a tentant-admin of any shared tenants. This does not cover the case of being an admin of another tenant.

It's possible that some query magic might be able to cover the case, but I haven't managed that yet.

I will work on this some more.

Reply:
I have separated the delete and update access check functions. The delete function checks for super-admin, and otherwise returns false. The update check function checks that the only affected tenants are ones that the requesting user is a tenant-admin of.